app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -129,31 +134,37 @@ class ThreadAdapter(
         return ThreadViewHolder(
             ItemMessageBinding.inflate(LayoutInflater.from(parent.context), parent, false),
             shouldLoadDistantResources,
+            isForPrinting,
             threadAdapterCallbacks?.onContactClicked,
             threadAdapterCallbacks?.onAttachmentClicked,
         )
     }
 
     override fun onBindViewHolder(holder: ThreadViewHolder, position: Int, payloads: MutableList<Any>) = runCatchingRealm {
         with(holder.binding) {
-            val payload = payloads.firstOrNull()
-            if (payload !is NotifyType) {
+            if (payloads.isEmpty()) {
                 super.onBindViewHolder(holder, position, payloads)
-                return
+                return@runCatchingRealm
```

Comment:
Do we really need to specify this? I feel like it makes lines way heavier to read for no real gain, when you see `return` you already know the outer function will exit. It's not like like it could be ambiguous.

Same for others in here

Reply:
@KevinBoulongne let's battle with @LunarX